modules/packages/UnitTest.chpl

```diff
@@ -242,6 +243,7 @@ Output:
 */
 module UnitTest {
   use Reflection;
+  use Regex;
```

Comment:
this is an unfortunate side effect, now all consumers of `UnitTest` require `CHPL_RE2=bundled`. Even if they don't use that. `mason` already has this requirement, but there could be other uses of `UnitTest` than `mason` (although I don't know any beyond our tests)

And I suspect this will causes all UnitTest tests (haha) to fail in the quickstart config, for example

Reply:
Hmm, that's something to think about. I'll bring this up with the team just to be on the safe side. My quick response is: you need `CHPL_RE2=bundled` to use a standard module, Regex. This is a package module. I think this is a acceptable dependency.

Reply:
I agree, its an acceptable dependency in my mind.

I bring it up because:
- its a changing requirement (that should be documented too)
- it will likely effect nightly tests

Reply:
Resolved via https://github.com/chapel-lang/chapel/pull/26981/commits/091243ee4bd2274de79996ab79c9e897803d3733